Mcp Server Module based on Java SDK (#11)
micronaut-projects/micronaut-mcp: src/main/docs/guide/server/primitives/prompts.adoc

____
Prompts: Pre-defined templates or instructions that guide language model interactions
____

> You can define prompts by defining methods annotated with api:mcp.annotations.Prompt[] in a `@Singleton` bean.

Reviewer: ```suggestion
The preferred way to define prompts is to annotate a method with api:mcp.annotations.Prompt[] in a `@Singleton` bean.
```

[source, Java]
----
include::micronaut-mcp-server/src/test/java/io/micronaut/mcp/server/stateless/sync/prompts/Prompts.java[tags="imports,clazz",indent=0]
----

> Alternatively, you define prompts registering beans (typically in a https://docs.micronaut.io/latest/guide/#factories[bean factory]).

Reviewer: ```suggestion
Alternatively, you define prompts registering beans (typically in a https://docs.micronaut.io/latest/guide/#factories[bean factory]) using the low-level MCP SDK API.
```
